bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/PanamaxFurmanAbstractHandler.java

```diff
@@ -0,0 +1,216 @@
+        }
+        if (System.currentTimeMillis() - REQUEST_LAST_SENT_AT.getOrDefault(commandToTransmitToThing,
+                0L) < AVOID_DUPLICATE_REQUESTS_WITHIN_MILLIS) {
+            logger.debug("Dropping request of '{}' since it was sent less than {}ms ago.  @{}",
+                    commandToTransmitToThing, AVOID_DUPLICATE_REQUESTS_WITHIN_MILLIS, getConnectionName());
+        } else {
+            REQUEST_LAST_SENT_AT.put(commandToTransmitToThing, System.currentTimeMillis());
+            connector.sendRequestToDevice(commandToTransmitToThing);
+        }
+    }
+
+    /**
+     * Invoked when the user wants to send a command to the Power Conditioner
+     */
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == RefreshType.REFRESH) {
+            requestStatusOf(channelUID.getId());
+        } else {
+            // Update command
+            String channelString = channelUID.getId();
+            Integer outletNumber = PanamaxFurmanAbstractHandler.getOutletFromChannelUID(channelString);
+            PowerConditionerChannel channel = PowerConditionerChannel.from(channelString);
+            if (channel.getStateClass() == null) {
+                logger.warn("Channel {} had null state class, command not sent  {}.  @{}", channel, command.getClass(),
+                        getConnectionName());
+                return;
+            } else if (!channel.getStateClass().isInstance(command)) {
+                logger.error("Update command {} NOT sent:  it expected State of class {} but was passed {}.  @{}",
+                        channel, channel.getStateClass(), command.getClass(), getConnectionName());
+                return;
+            }
+            State stateToSet = channel.getStateClass().cast(command);
+            String textToTransmitToThing = protocolMapper.buildUpdateString(channel, outletNumber, stateToSet);
+            connector.sendRequestToDevice(textToTransmitToThing);
+        }
+    }
+
+    @Override
+    public void onConnectivityEvent(PanamaxFurmanConnectivityEvent event) {
+        // Only invoke updateStatus if it has actually changed
+        if (event instanceof PanamaxFurmanConnectionActiveEvent) {
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } else if (event instanceof PanamaxFurmanConnectionBrokenEvent
+                && getThing().getStatus() != ThingStatus.OFFLINE) {
+            String errorDetail = ((PanamaxFurmanConnectionBrokenEvent) event).getErrorDetail();
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorDetail);
+        } else {
+            logger.warn("Unhandled PanamaxFurmanConnectivityEvent of {}", event.getClass());
+        }
+    }
+
+    @Override
+    public void onInformationReceived(String data) {
+        @Nullable
```

Comment:
inline nullable annotation should not be needed as the compilter should detect it.

Reply:
that's what I thought but my local build kept showing the warnings.  let me make another update and push it for the PR build

Reply:
null anotations are not perfect, this warning is ok.